infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -780,9 +781,23 @@ private LogData  readRecord(SegmentHandle sh, long address)
                             EnumSet.of(StandardOpenOption.READ));
                 }
             } else {
-                return FileChannel.open(FileSystems.getDefault().getPath(filePath),
+                boolean newFile = false;
+                
+                File segFile = new File(filePath);
```

Comment:
This seems like it could be cleaned up. Perhaps try opening the file, and creating it if it fails, syncing the dirent as well.

Reply:
Yea, that is cleaner. Done.